fluttercommunity/flutter_downloader
# [iOS] allow background download when app is not running

This PR allows downloading files in background when app is not running. Currently the plugin cancels all download tasks when the app is terminated. For info about background download on iOS, check out the doc [here](https://developer.apple.com/documentation/foundation/url_loading_system/downloading_files_in_the_background?language=objc). 

This PR also fixes the follow issues: 
https://github.com/fluttercommunity/flutter_downloader/issues/675
https://github.com/fluttercommunity/flutter_downloader/issues/663

Steps for testing background download:

1. Build and install the example app on an iOS simulator or device
2. Run app from Home screen (background download might not work properly if the app has been started from xcode with a debugger)
3. start downloading a file
4. tap the Exit button on the top nav to exit the app gracefully. Do not kill app using task manager otherwise the any download tasks will be cancelled by the OS
5. wait a short while
6. open the example app again from the home screen
7. the downloading file should be completed with a “Ready” status

video demo:

https://user-images.githubusercontent.com/829645/176210370-9105153b-c104-4216-b971-2bf83baedd36.mov
